app/src/androidTest/java/com/duckduckgo/app/trackerdetection/TrackerDetectorInstrumentationTest.kt

```diff
@@ -16,62 +16,107 @@
 
 package com.duckduckgo.app.trackerdetection
 
-
-import android.support.test.runner.AndroidJUnit4
-import com.duckduckgo.app.trackerdetection.Client.ClientName.EASYLIST
-import com.duckduckgo.app.trackerdetection.Client.ClientName.EASYPRIVACY
+import com.duckduckgo.app.trackerdetection.model.DisconnectTracker
+import com.duckduckgo.app.trackerdetection.model.NetworkTrackers
 import com.duckduckgo.app.trackerdetection.model.ResourceType
+import com.nhaarman.mockito_kotlin.any
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.whenever
 import org.junit.Assert.assertFalse
 import org.junit.Assert.assertTrue
-import org.junit.Before
 import org.junit.Test
-import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers.anyString
+import java.util.*
 
 
-@RunWith(AndroidJUnit4::class)
 class TrackerDetectorInstrumentationTest {
 
+    private val networkTrackers = NetworkTrackers()
+    private val trackerDetector = TrackerDetector(networkTrackers)
+
     companion object {
-        private val documentUrl = "http://example.com"
         private val resourceType = ResourceType.UNKNOWN
```

Comment:
What's the thinking with resourceType? It's passed in as a parameter, but never used (for instance, if I change the code to ignore the passed in value and use ResourceType.IMAGE the tests still pass.

Is it really needed?

Reply:
Good question. It is needed however it is not to relevant at this level, if it were a more complex type I would mock it. It is passed to the tracker clients and adblock specifically needs it to match rules correctly (fyi the problems that iOS and the extension are having with the tacker detection are around this area). 

The ResourceType itself is tested in ResourceTypeInstrumentationTest, the current behaviour there is an improved version of what brave has on iOS although we may expand and improve on it over time. An integration test suite (like the one the extension team are writing) would be a good addition for both android and ios to validate that the third party components we're using are working correctly. That goes well beyond the scope of this PR so I'll add a separate task for that.